lib/rules/unit-no-unknown/index.js

```diff
@@ -22,10 +22,6 @@ const messages = ruleMessages(ruleName, {
 	rejected: (unit) => `Unexpected unknown unit "${unit}"`,
 });
 
-// The map property name (in map cleared from comments and spaces) always
-// has index that being divided by 4 gives remainder equals 0
-const mapPropertyNameIndexOffset = 4;
```

Comment:
[question] This `mapPropertyNameIndexOffset` constant (and also [`isMap`](https://github.com/stylelint/stylelint/blob/5d98cc3e65f5fc1551dc2d1166efb481e3bba679/lib/utils/isMap.js)) was introduced with PR #4450. I'm not sure the context, but it seems better to me to improve `isStandardSyntaxDeclaration` as @jeddy3 suggested on <https://github.com/stylelint/stylelint/issues/5434#issuecomment-894419800>, instead of changing the current logic.

What do you think?

Reply:
@ybiquitous this PR doesn't change current logic. It just changes the assumption, that the mapPropertyNameIndexOffset is always 4 - because it isn't, when map value has "*" inside. I've changed it, that now we check that property ends with ":", which is much more reliable in my opinion. 

If we'll use isStandardSyntaxDeclaration it will disable this rule for all scss maps. Which I guess we don't want to do, because now it works ok, instead of the edge case with "*" in value. And for sure we can't make this kind of logic in 119 line, since upper "if" is `if (unit.toLowerCase() === 'x') ` (what with the other units?)

What do you think @jeddy3, am I thinking correctly?

Reply:
@sylwia-lask Thanks for your pull request. 

Let's remove the map functionality, including `isMap`, as it's inconsistent with how all the other built-in rules deal with non-standard syntax, i.e. by ignoring them using the `isStandardSyntax*` utils. Code for non-standard syntax only belongs in the `isStandardSyntax*` utils.

The built-in rules are for standard CSS constructs, and so this rule should ignore maps.

Someone can then write a `scss/unit-no-unknown` plugin for [`stylelint-scss`](https://github.com/kristerkari/stylelint-scss) that specifically checks for unknown units in Sass constructs like maps. The plugin can deal with the awkward edge cases that come from working with SCSS syntax using parsers designed for CSS). The plugin would be used in tandem with the built-in rule.

Ref: https://github.com/stylelint/stylelint/pull/5454#pullrequestreview-728372653

Reply:
@jeddy3 in test file for unit-no-unknown are lots of tests for less, sass and scss syntaxes. Should I remove it?

Reply:
Let's just remove the `reject` tests that will fail when we ignore the constructs outright. The `accept` tests will remain valid as they'll now test that the constructs are ignored.

When you remove the map code and add the (amended) `isStandardSyntaxDeclaration` util, it should become clear which tests these are.